// @corellium/corellium-cli/src/commands/projects/vpnConfig.js
const { handleError } = require('../../error');
const fs = require('fs');
const fsp = fs.promises;
const { getCorelliumApi } = require('../../corellium-api');

async function builder (yargs) {
  yargs
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    })
    .option('project', {
      type: 'input',
      describe: 'Id of the project to pull ovpn configuration from.',
      string: true,
    })
    .option('path', {
      type: 'input',
      describe: 'Path to where to save the ovpn configuration file.',
      string: true,
    });
}

async function handler (argv) {
  const project = argv.project;
  const corellium = await getCorelliumApi();
  const projectApi = await corellium.getProject(project);
  const savePath = argv.path;
  try {
    const vpnConfigBuffer = await projectApi.vpnConfig();
    await fsp.writeFile(savePath, vpnConfigBuffer.toString());
  } catch (err) {
    handleError(err, 'vpnConfig failed', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'vpnConfig [project] [path]',
  describe: 'Download Openvpn config for a project',
};
