// @corellium/corellium-cli/src/commands/instances/create.js
const { handleError } = require('../../error');
const { getApi, waitForState, waitForAgentReady } = require('../../utils');
const AgentCLI = require('../../clients/Agent');

async function builder (yargs) {
  yargs
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    })
    .option('format', {
      type: 'input',
      describe: 'Output format (default is json) e.g. table',
      string: true,
      choices: ['table', 'json', 'csv'],
    })
    .positional('flavor', {
      type: 'string',
      describe: 'Flavor',
      demandOption: true,
    })
    .positional('os', {
      type: 'string',
      describe: 'Operating system',
      demandOption: true,
    })
    .positional('project', {
      type: 'string',
      describe: 'Project ID to associate with instance',
      demandOption: true,
    })
    .option('os-build', {
      type: 'string',
      describe: 'OS Build',
      demandOption: false,
    })
    .option('name', {
      type: 'string',
      describe: 'Optional name of instance',
      demandOption: false,
    })
    .option('wait', {
      type: 'boolean',
      describe: 'Wait for the instance to be ready',
      demandOption: false,
    })
    .option('dyld-aslr', {
      type: 'boolean',
      describe: 'Enable ASLR for DYLD Shared Cache. (--no-dyld-aslr to disable)',
      default: true,
      demandOption: false,
    })
    .option('app-aslr', {
      type: 'boolean',
      describe: 'Enable ASLR for Applications. (--no-app-aslr to disable)',
      default: true,
      demandOption: false,
    });
}

async function handler (argv) {
  const flavor = argv.flavor;
  const os = argv.os;
  const project = argv.project;
  const name = argv.name;
  const api = await getApi();
  const bootOptions = {};
  const additionalTags = [];
  const instanceOptions = { flavor, os, project, name };

  if (argv['app-aslr'] === false) {
    additionalTags.push('noappaslr');
  }

  if (argv['dyld-aslr'] === false) {
    additionalTags.push('nodyldaslr');
  }

  // only define additional tags if any are present
  if (additionalTags.length > 0) {
    bootOptions.additionalTags = additionalTags;
  }

  // only define bootOptions if it contains any values
  if (Object.keys(bootOptions).length > 0) {
    instanceOptions.bootOptions = bootOptions;
  }

  if (argv['os-build']) {
    instanceOptions.osbuild = argv['os-build'];
  }

  try {
    const instance = await api.v1CreateInstance(instanceOptions);
    // Wait 360 attempts for the instance to be created (30 minutes)
    if (argv.wait) {
      await waitForState(instance.id, api.v1GetInstance.bind(api), 'on', 360);
      const agent = new AgentCLI({ instance: instance.id });
      await waitForAgentReady(agent, 360);
    }
    console.log(instance.id);
  } catch (err) {
    handleError(err, 'Failed to create instance', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'create <flavor> <os> <project> [name] [wait]',
  describe: 'Create instance and associate with a project',
};
