// @corellium/corellium-cli/src/commands/instances/file.js
const AgentCLI = require('../../clients/Agent');
const { displayTable } = require('../../table');
const { validateNonEmpty } = require('../../utils');
const { handleError } = require('../../error');

async function builder (yargs) {
  yargs
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    })
    .option('project', {
      type: 'input',
      describe: 'Project ID',
      demandOption: true,
      string: true,
      validateNonEmpty,
    })
    .option('instance', {
      type: 'input',
      describe: 'Instance ID',
      demandOption: true,
      string: true,
      validateNonEmpty,
    })
    .option('format', {
      type: 'input',
      describe: 'Output format (default is json) e.g. table',
      string: true,
      choices: ['table', 'json', 'csv'],
    });
}

async function handler (argv) {
  try {
    const agent = new AgentCLI(argv);
    const files = await agent.listFiles();
    await displayTable(argv.format, files, ['name', 'size']);
  } catch (err) {
    handleError(err, 'List files failed', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'files',
  describe: 'List files on device instance',
};
