// @corellium/corellium-cli/src/commands/instances/get.js
const { handleError } = require('../../error');
const { displayTable } = require('../../table');
const { getApi } = require('../../utils');

async function builder (yargs) {
  yargs
    .option('instance', {
      alias: 'i',
      type: 'string',
      demandOption: true,
      describe: 'virtual device id',
    })
    .option('format', {
      type: 'input',
      describe: 'Output format (default is json) e.g. table',
      string: true,
      choices: ['table', 'json', 'csv'],
      default: 'json',
    })
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    });
}

async function handler (argv) {
  const api = await getApi();

  try {
    const instance = await api.v1GetInstance(argv.instance);
    if (!instance) {
      throw new Error(`Instance ${argv.instance} not found`);
    }
    await displayTable(
      argv.format,
      argv.format === 'json' ? instance : [instance],
      ['id', 'flavor', 'type', 'state', 'os'],
      [
        { column: 3, width: 5 },
        { column: 2, width: 4 },
        { column: 1, width: 6 },
      ],
    );
  } catch (err) {
    handleError(err, 'Get instance failed', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'get',
  describe: 'Get instance',
};
