// @corellium/corellium-cli/src/commands/instances/list.js
const { handleError } = require('../../error');
const { displayTable } = require('../../table');
const { getApi } = require('../../utils');

async function builder (yargs) {
  yargs
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    })
    .option('project', {
      type: 'input',
      describe: 'Optionally filter by project name',
    })
    .option('format', {
      type: 'input',
      describe: 'Output format (default is json) e.g. table',
      string: true,
      choices: ['table', 'json', 'csv'],
      default: 'json',
    });
}

async function handler (argv) {
  const api = await getApi();
  try {
    let instances;
    if (argv.project) {
      instances = await api.v1GetProjectInstances(argv.project, {});
    } else {
      instances = await api.v1GetInstances();
    }
    if (!instances.length && argv.format !== 'json') {
      console.error('Found no matching Instances');
    }
    await displayTable(
      argv.format,
      instances,
      ['id', 'flavor', 'type', 'state', 'os'],
      [
        { column: 3, width: 5 },
        { column: 2, width: 4 },
        { column: 1, width: 6 },
      ],
    );
  } catch (err) {
    handleError(err, 'List failed', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'list',
  aliases: ['ls'],
  describe: 'List instances',
};
