// @corellium/corellium-cli/src/commands/instances/unlock.js
// this is for iOS only

const chalk = require('chalk');
const AgentCLI = require('../../clients/Agent');
const { handleError } = require('../../error');
const { getApi } = require('../../utils');

async function builder (yargs) {
  yargs
    .option('instance', {
      alias: 'i',
      type: 'string',
      demandOption: true,
      describe: 'virtual device id',
    })
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    });
}

async function handler (argv) {
  const api = await getApi();

  try {
    // verify instance type
    const instance = await api.v1GetInstance(argv.instance);
    if (!instance) {
      throw new Error(`Instance ${argv.instance} not found`);
    }
    if (instance.type !== 'ios') {
      throw new Error(`Instance ${argv.instance} is not an iOS device`);
    }

    const agent = new AgentCLI(argv);
    await agent.unlock();
    console.log(chalk.green(`Successfully unlocked ${argv.instance}`));
  } catch (err) {
    handleError(err, 'Unlock failed', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'unlock',
  describe: 'Unlock iOS device',
};
