// @corellium/corellium-cli/src/commands/webplayer/create.js
const Webplayer = require('../../clients/Webplayer');
const { displayTable } = require('../../table');
const { validateNonEmpty } = require('../../utils');
const { handleError } = require('../../error');

async function builder (yargs) {
  yargs
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    })
    .option('project', {
      type: 'input',
      describe: 'Project ID',
      demandOption: true,
      string: true,
      validateNonEmpty,
    })
    .option('instance', {
      type: 'input',
      describe: 'Instance ID',
      demandOption: true,
      string: true,
      validateNonEmpty,
    })
    .option('features', {
      type: 'input',
      describe: 'Enabled Feature Set',
      demandOption: true,
      string: true,
      validateNonEmpty,
    })
    .option('permissions', {
      type: 'input',
      describe: 'Endpoint Permissions',
      demandOption: false,
      string: true,
      validateNonEmpty,
    })
    .option('expiresIn', {
      type: 'input',
      describe: 'Token expiration in seconds',
      demandOption: true,
      string: true,
      validateNonEmpty,
    })
    .option('format', {
      type: 'input',
      describe: 'Output format (default is json) e.g. table',
      string: true,
      choices: ['table', 'json', 'csv'],
    });
}

async function handler (argv) {
  try {
    const webplayer = new Webplayer(argv);
    const session = await webplayer.createSession();
    // log.info('Created Webplayer session')
    await displayTable(argv.format || 'json', [session], ['identifier', 'token', 'expiration', 'features']);
  } catch (err) {
    handleError(err, 'Create Webplayer session failed', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'create',
  describe: 'Create a Webplayer session returning the URL and token',
};
