// @corellium/corellium-cli/src/commands/mast/delete-assessment.js
const Client = require('../../clients/Client');
const { handleError } = require('../../error');
const log = require('../../logging');
const { MAST_API_BASE_PATH } = require('./constants');

/**
 * Delete an assessment via corellium-mast API.
 *
 * @param yargs
 * @returns {Promise<void>}
 */
async function builder (yargs) {
  yargs
    .option('assessment', {
      alias: 'a',
      type: 'string',
      demandOption: true,
      describe: 'MATRIX assessment id',
    })
    .option('instance', {
      alias: 'i',
      type: 'string',
      demandOption: true,
      describe: 'virtual device id',
    })
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'console will receive verbose error output',
      default: false,
    });
}

async function handler (argv) {
  const { assessment, instance, verbose } = argv;
  const client = new Client(argv);
  try {
    await client._fetch('DELETE', `${MAST_API_BASE_PATH}/${instance}/assessments/${assessment}`);
    log.info('Assessment deleted.');
  } catch (error) {
    handleError(error, 'delete-assessment failed', verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'delete-assessment',
  describe: 'Delete MATRIX assessment',
};
