// @corellium/corellium-cli/src/commands/mast/download-report.js
const Client = require('../../clients/Client');
const { handleError } = require('../../error');
const { MAST_API_BASE_PATH } = require('./constants');

/**
 * Download assessment report via corellium-mast API.
 *
 * @param yargs
 * @returns {Promise<void>}
 */
async function builder (yargs) {
  yargs
    .option('assessment', {
      alias: 'a',
      type: 'string',
      demandOption: true,
      describe: 'MATRIX assessment id',
    })
    // NOTE: only required due to coordinator proxy needing to authenticate access to instance
    .option('instance', {
      alias: 'i',
      type: 'string',
      demandOption: true,
      describe: 'virtual device id',
      default: process.env.INSTANCE,
    })
    .option('format', {
      alias: 'f',
      type: 'string',
      describe: 'The desired format of the report.',
      choices: ['html', 'json'],
      default: 'html',
    })
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'console will receive verbose error output',
      default: false,
    });
}

async function handler (argv) {
  const { assessment, instance, format, verbose } = argv;
  const client = new Client(argv);
  try {
    const res = await client._fetch(
      'GET',
      `${MAST_API_BASE_PATH}/${instance}/assessments/${assessment}/download?format=${format}`,
    );
    console.log(format === 'json' ? JSON.stringify(res, null, 2) : res); // print deep json
  } catch (error) {
    handleError(error, 'download-report failed', verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'download-report',
  describe: 'Download MATRIX assessment report',
};
