// @corellium/corellium-cli/src/commands/mast/get-assessment.js
const Client = require('../../clients/Client');
const { handleError } = require('../../error');
const { MAST_API_BASE_PATH } = require('./constants');

/**
 * Get an assessment via corellium-mast API.
 *
 * @param yargs
 * @returns {Promise<void>}
 */
async function builder (yargs) {
  yargs
    .option('assessment', {
      alias: 'a',
      type: 'string',
      demandOption: true,
      describe: 'MATRIX assessment id',
    })
    // NOTE: only required due to coordinator proxy needing to authenticate access to instance
    .option('instance', {
      alias: 'i',
      type: 'string',
      demandOption: true,
      describe: 'virtual device id',
      default: process.env.INSTANCE,
    })
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'console will receive verbose error output',
      default: false,
    });
}

async function handler (argv) {
  const { assessment, instance, verbose } = argv;
  const client = new Client(argv);
  try {
    const res = await client._fetch('GET', `${MAST_API_BASE_PATH}/${instance}/assessments/${assessment}`);
    console.log(JSON.stringify(res));
  } catch (error) {
    handleError(error, 'get-assessment failed', verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'get-assessment',
  describe: 'Get MATRIX assessment',
};
