// @corellium/corellium-cli/src/commands/login.js
const chalk = require('chalk');
const { updateProfile, profilePath } = require('../profile');
const { handleError } = require('../error');
const { validateNonEmpty, getCACert } = require('../utils');
const https = require('https');
const axios = require('axios').default;
const prompts = require('prompts');

async function builder (yargs) {
  yargs
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    })
    .option('endpoint', {
      type: 'input',
      describe: 'Server endpoint e.g. https://app.corellium.com',
      demandOption: true,
      string: true,
      default: process.env.ENDPOINT,
      validateNonEmpty,
    })
    .option('username', {
      type: 'input',
      describe: 'Username',
      demandOption: false,
      string: true,
      default: process.env.USERNAME,
      validateNonEmpty,
    })
    .option('password', {
      type: 'password',
      describe: 'Password',
      demandOption: false,
      string: true,
      prompt: 'if-no-arg',
      default: process.env.PASSWORD,
      validateNonEmpty,
    })
    .option('apitoken', {
      type: 'password',
      describe: 'API Token',
      demandOption: false,
      string: true,
      validateNonEmpty,
    });
}

async function handler (argv) {
  const { username, apitoken, endpoint } = argv;
  let { password } = argv;

  if (username && !password) {
    password = (
      await prompts({
        message: 'Enter your password',
        name: 'password',
        type: 'text',
        style: 'invisible',
      })
    ).password;
  }

  const sanitizedEndpoint = endpoint.slice(-1) === '/' ? endpoint.slice(0, -1) : endpoint;

  // We need a special agent just for login because we can't assume profile has been written yet
  const CACert = await getCACert({ endpoint: sanitizedEndpoint });
  axios.defaults.httpsAgent = new https.Agent({ ca: CACert });

  try {
    let data;

    if (apitoken) {
      data = (await axios.post(`${sanitizedEndpoint}/api/v1/tokens`, { apiToken: apitoken })).data;
    } else {
      data = (await axios.post(`${sanitizedEndpoint}/api/v1/tokens`, { username, password })).data;
    }
    await updateProfile({ token: data.token, expiration: data.expiration, endpoint: sanitizedEndpoint });
    console.log(chalk.green(`Login successful, profile saved to ${profilePath}`));
  } catch (err) {
    handleError(err, 'Login failed', argv.v);
  }
}

module.exports = {
  builder,
  handler,
  command: 'login',
  describe:
    'Login and save authentication token to local profile.\r\nOptionally supply an API Token instead of credentials.',
};
