// @corellium/client-api/dist/model/AgentApp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AgentApp model module.
 * @module model/AgentApp
 * @version 0.3.2
 */
var AgentApp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AgentApp</code>.
   * 
   * @alias module:model/AgentApp
   */
  function AgentApp() {
    _classCallCheck(this, AgentApp);
    AgentApp.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AgentApp, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AgentApp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AgentApp} obj Optional instance to populate.
     * @return {module:model/AgentApp} The populated <code>AgentApp</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AgentApp();
        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }
        if (data.hasOwnProperty('running')) {
          obj['running'] = _ApiClient["default"].convertToType(data['running'], 'Boolean');
        }
        if (data.hasOwnProperty('diskUsage')) {
          obj['diskUsage'] = _ApiClient["default"].convertToType(data['diskUsage'], 'Number');
        }
        if (data.hasOwnProperty('Date')) {
          obj['Date'] = _ApiClient["default"].convertToType(data['Date'], 'Number');
        }
        if (data.hasOwnProperty('applicationType')) {
          obj['applicationType'] = _ApiClient["default"].convertToType(data['applicationType'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('bundleID')) {
          obj['bundleID'] = _ApiClient["default"].convertToType(data['bundleID'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>AgentApp</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AgentApp</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      }
      // ensure the json data is a string
      if (data['applicationType'] && !(typeof data['applicationType'] === 'string' || data['applicationType'] instanceof String)) {
        throw new Error("Expected the field `applicationType` to be a primitive type in the JSON string but got " + data['applicationType']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['bundleID'] && !(typeof data['bundleID'] === 'string' || data['bundleID'] instanceof String)) {
        throw new Error("Expected the field `bundleID` to be a primitive type in the JSON string but got " + data['bundleID']);
      }
      return true;
    }
  }]);
  return AgentApp;
}();
/**
 * 
 * @member {Array.<String>} tags
 */
AgentApp.prototype['tags'] = undefined;

/**
 * 
 * @member {Boolean} running
 */
AgentApp.prototype['running'] = undefined;

/**
 * 
 * @member {Number} diskUsage
 */
AgentApp.prototype['diskUsage'] = undefined;

/**
 * 
 * @member {Number} Date
 */
AgentApp.prototype['Date'] = undefined;

/**
 * 
 * @member {String} applicationType
 */
AgentApp.prototype['applicationType'] = undefined;

/**
 * 
 * @member {String} name
 */
AgentApp.prototype['name'] = undefined;

/**
 * 
 * @member {String} bundleID
 */
AgentApp.prototype['bundleID'] = undefined;
var _default = AgentApp;
exports["default"] = _default;
