// @corellium/client-api/dist/model/Firmware.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Firmware model module.
 * @module model/Firmware
 * @version 0.3.2
 */
var Firmware = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Firmware</code>.
   * 
   * @alias module:model/Firmware
   */
  function Firmware() {
    _classCallCheck(this, Firmware);
    Firmware.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Firmware, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Firmware</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Firmware} obj Optional instance to populate.
     * @return {module:model/Firmware} The populated <code>Firmware</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Firmware();
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'String');
        }
        if (data.hasOwnProperty('buildid')) {
          obj['buildid'] = _ApiClient["default"].convertToType(data['buildid'], 'String');
        }
        if (data.hasOwnProperty('AndroidFlavor')) {
          obj['AndroidFlavor'] = _ApiClient["default"].convertToType(data['AndroidFlavor'], 'String');
        }
        if (data.hasOwnProperty('APIVersion')) {
          obj['APIVersion'] = _ApiClient["default"].convertToType(data['APIVersion'], 'String');
        }
        if (data.hasOwnProperty('sha256sum')) {
          obj['sha256sum'] = _ApiClient["default"].convertToType(data['sha256sum'], 'String');
        }
        if (data.hasOwnProperty('sha1sum')) {
          obj['sha1sum'] = _ApiClient["default"].convertToType(data['sha1sum'], 'String');
        }
        if (data.hasOwnProperty('md5sum')) {
          obj['md5sum'] = _ApiClient["default"].convertToType(data['md5sum'], 'String');
        }
        if (data.hasOwnProperty('size')) {
          obj['size'] = _ApiClient["default"].convertToType(data['size'], 'Number');
        }
        if (data.hasOwnProperty('uniqueId')) {
          obj['uniqueId'] = _ApiClient["default"].convertToType(data['uniqueId'], 'String');
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
        if (data.hasOwnProperty('releasedate')) {
          obj['releasedate'] = _ApiClient["default"].convertToType(data['releasedate'], 'Date');
        }
        if (data.hasOwnProperty('uploaddate')) {
          obj['uploaddate'] = _ApiClient["default"].convertToType(data['uploaddate'], 'Date');
        }
        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
        if (data.hasOwnProperty('orig_url')) {
          obj['orig_url'] = _ApiClient["default"].convertToType(data['orig_url'], 'String');
        }
        if (data.hasOwnProperty('filename')) {
          obj['filename'] = _ApiClient["default"].convertToType(data['filename'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Firmware</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Firmware</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['version'] && !(typeof data['version'] === 'string' || data['version'] instanceof String)) {
        throw new Error("Expected the field `version` to be a primitive type in the JSON string but got " + data['version']);
      }
      // ensure the json data is a string
      if (data['buildid'] && !(typeof data['buildid'] === 'string' || data['buildid'] instanceof String)) {
        throw new Error("Expected the field `buildid` to be a primitive type in the JSON string but got " + data['buildid']);
      }
      // ensure the json data is a string
      if (data['AndroidFlavor'] && !(typeof data['AndroidFlavor'] === 'string' || data['AndroidFlavor'] instanceof String)) {
        throw new Error("Expected the field `AndroidFlavor` to be a primitive type in the JSON string but got " + data['AndroidFlavor']);
      }
      // ensure the json data is a string
      if (data['APIVersion'] && !(typeof data['APIVersion'] === 'string' || data['APIVersion'] instanceof String)) {
        throw new Error("Expected the field `APIVersion` to be a primitive type in the JSON string but got " + data['APIVersion']);
      }
      // ensure the json data is a string
      if (data['sha256sum'] && !(typeof data['sha256sum'] === 'string' || data['sha256sum'] instanceof String)) {
        throw new Error("Expected the field `sha256sum` to be a primitive type in the JSON string but got " + data['sha256sum']);
      }
      // ensure the json data is a string
      if (data['sha1sum'] && !(typeof data['sha1sum'] === 'string' || data['sha1sum'] instanceof String)) {
        throw new Error("Expected the field `sha1sum` to be a primitive type in the JSON string but got " + data['sha1sum']);
      }
      // ensure the json data is a string
      if (data['md5sum'] && !(typeof data['md5sum'] === 'string' || data['md5sum'] instanceof String)) {
        throw new Error("Expected the field `md5sum` to be a primitive type in the JSON string but got " + data['md5sum']);
      }
      // ensure the json data is a string
      if (data['uniqueId'] && !(typeof data['uniqueId'] === 'string' || data['uniqueId'] instanceof String)) {
        throw new Error("Expected the field `uniqueId` to be a primitive type in the JSON string but got " + data['uniqueId']);
      }
      // ensure the json data is a string
      if (data['url'] && !(typeof data['url'] === 'string' || data['url'] instanceof String)) {
        throw new Error("Expected the field `url` to be a primitive type in the JSON string but got " + data['url']);
      }
      // ensure the json data is a string
      if (data['orig_url'] && !(typeof data['orig_url'] === 'string' || data['orig_url'] instanceof String)) {
        throw new Error("Expected the field `orig_url` to be a primitive type in the JSON string but got " + data['orig_url']);
      }
      // ensure the json data is a string
      if (data['filename'] && !(typeof data['filename'] === 'string' || data['filename'] instanceof String)) {
        throw new Error("Expected the field `filename` to be a primitive type in the JSON string but got " + data['filename']);
      }
      return true;
    }
  }]);
  return Firmware;
}();
/**
 * 
 * @member {String} version
 */
Firmware.prototype['version'] = undefined;

/**
 * 
 * @member {String} buildid
 */
Firmware.prototype['buildid'] = undefined;

/**
 * Android only flavor
 * @member {String} AndroidFlavor
 */
Firmware.prototype['AndroidFlavor'] = undefined;

/**
 * Android only API version
 * @member {String} APIVersion
 */
Firmware.prototype['APIVersion'] = undefined;

/**
 * 
 * @member {String} sha256sum
 */
Firmware.prototype['sha256sum'] = undefined;

/**
 * 
 * @member {String} sha1sum
 */
Firmware.prototype['sha1sum'] = undefined;

/**
 * 
 * @member {String} md5sum
 */
Firmware.prototype['md5sum'] = undefined;

/**
 * 
 * @member {Number} size
 */
Firmware.prototype['size'] = undefined;

/**
 * 
 * @member {String} uniqueId
 */
Firmware.prototype['uniqueId'] = undefined;

/**
 * Firmware metadata
 * @member {Object} metadata
 */
Firmware.prototype['metadata'] = undefined;

/**
 * Release Date
 * @member {Date} releasedate
 */
Firmware.prototype['releasedate'] = undefined;

/**
 * Date uploaded
 * @member {Date} uploaddate
 */
Firmware.prototype['uploaddate'] = undefined;

/**
 * URL firmware is available at
 * @member {String} url
 */
Firmware.prototype['url'] = undefined;

/**
 * URL firmware is available at from vendor
 * @member {String} orig_url
 */
Firmware.prototype['orig_url'] = undefined;

/**
 * 
 * @member {String} filename
 */
Firmware.prototype['filename'] = undefined;
var _default = Firmware;
exports["default"] = _default;
