// @corellium/client-api/dist/model/InstanceCreateOptions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _InstanceBootOptions = _interopRequireDefault(require("./InstanceBootOptions"));
var _Model = _interopRequireDefault(require("./Model"));
var _VolumeOptions = _interopRequireDefault(require("./VolumeOptions"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The InstanceCreateOptions model module.
 * @module model/InstanceCreateOptions
 * @version 0.3.2
 */
var InstanceCreateOptions = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InstanceCreateOptions</code>.
   * 
   * @alias module:model/InstanceCreateOptions
   * @param flavor {String} the flavor id
   * @param project {String} project UUID
   * @param os {String} OS Version
   */
  function InstanceCreateOptions(flavor, project, os) {
    _classCallCheck(this, InstanceCreateOptions);
    InstanceCreateOptions.initialize(this, flavor, project, os);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(InstanceCreateOptions, null, [{
    key: "initialize",
    value: function initialize(obj, flavor, project, os) {
      obj['flavor'] = flavor;
      obj['project'] = project;
      obj['os'] = os;
    }

    /**
     * Constructs a <code>InstanceCreateOptions</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InstanceCreateOptions} obj Optional instance to populate.
     * @return {module:model/InstanceCreateOptions} The populated <code>InstanceCreateOptions</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InstanceCreateOptions();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }
        if (data.hasOwnProperty('flavor')) {
          obj['flavor'] = _ApiClient["default"].convertToType(data['flavor'], 'String');
        }
        if (data.hasOwnProperty('project')) {
          obj['project'] = _ApiClient["default"].convertToType(data['project'], 'String');
        }
        if (data.hasOwnProperty('os')) {
          obj['os'] = _ApiClient["default"].convertToType(data['os'], 'String');
        }
        if (data.hasOwnProperty('osbuild')) {
          obj['osbuild'] = _ApiClient["default"].convertToType(data['osbuild'], 'String');
        }
        if (data.hasOwnProperty('patches')) {
          obj['patches'] = _ApiClient["default"].convertToType(data['patches'], ['String']);
        }
        if (data.hasOwnProperty('fwpackage')) {
          obj['fwpackage'] = _ApiClient["default"].convertToType(data['fwpackage'], 'String');
        }
        if (data.hasOwnProperty('origFwPackageUrl')) {
          obj['origFwPackageUrl'] = _ApiClient["default"].convertToType(data['origFwPackageUrl'], 'String');
        }
        if (data.hasOwnProperty('encrypt')) {
          obj['encrypt'] = _ApiClient["default"].convertToType(data['encrypt'], 'Boolean');
        }
        if (data.hasOwnProperty('overrideWifiMAC')) {
          obj['overrideWifiMAC'] = _ApiClient["default"].convertToType(data['overrideWifiMAC'], 'String');
        }
        if (data.hasOwnProperty('volume')) {
          obj['volume'] = _VolumeOptions["default"].constructFromObject(data['volume']);
        }
        if (data.hasOwnProperty('snapshot')) {
          obj['snapshot'] = _ApiClient["default"].convertToType(data['snapshot'], 'String');
        }
        if (data.hasOwnProperty('bootOptions')) {
          obj['bootOptions'] = _InstanceBootOptions["default"].constructFromObject(data['bootOptions']);
        }
        if (data.hasOwnProperty('device')) {
          obj['device'] = _Model["default"].constructFromObject(data['device']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>InstanceCreateOptions</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>InstanceCreateOptions</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(InstanceCreateOptions.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      }
      // ensure the json data is a string
      if (data['flavor'] && !(typeof data['flavor'] === 'string' || data['flavor'] instanceof String)) {
        throw new Error("Expected the field `flavor` to be a primitive type in the JSON string but got " + data['flavor']);
      }
      // ensure the json data is a string
      if (data['project'] && !(typeof data['project'] === 'string' || data['project'] instanceof String)) {
        throw new Error("Expected the field `project` to be a primitive type in the JSON string but got " + data['project']);
      }
      // ensure the json data is a string
      if (data['os'] && !(typeof data['os'] === 'string' || data['os'] instanceof String)) {
        throw new Error("Expected the field `os` to be a primitive type in the JSON string but got " + data['os']);
      }
      // ensure the json data is a string
      if (data['osbuild'] && !(typeof data['osbuild'] === 'string' || data['osbuild'] instanceof String)) {
        throw new Error("Expected the field `osbuild` to be a primitive type in the JSON string but got " + data['osbuild']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['patches'])) {
        throw new Error("Expected the field `patches` to be an array in the JSON data but got " + data['patches']);
      }
      // ensure the json data is a string
      if (data['fwpackage'] && !(typeof data['fwpackage'] === 'string' || data['fwpackage'] instanceof String)) {
        throw new Error("Expected the field `fwpackage` to be a primitive type in the JSON string but got " + data['fwpackage']);
      }
      // ensure the json data is a string
      if (data['origFwPackageUrl'] && !(typeof data['origFwPackageUrl'] === 'string' || data['origFwPackageUrl'] instanceof String)) {
        throw new Error("Expected the field `origFwPackageUrl` to be a primitive type in the JSON string but got " + data['origFwPackageUrl']);
      }
      // ensure the json data is a string
      if (data['overrideWifiMAC'] && !(typeof data['overrideWifiMAC'] === 'string' || data['overrideWifiMAC'] instanceof String)) {
        throw new Error("Expected the field `overrideWifiMAC` to be a primitive type in the JSON string but got " + data['overrideWifiMAC']);
      }
      // validate the optional field `volume`
      if (data['volume']) {
        // data not null
        _VolumeOptions["default"].validateJSON(data['volume']);
      }
      // ensure the json data is a string
      if (data['snapshot'] && !(typeof data['snapshot'] === 'string' || data['snapshot'] instanceof String)) {
        throw new Error("Expected the field `snapshot` to be a primitive type in the JSON string but got " + data['snapshot']);
      }
      // validate the optional field `bootOptions`
      if (data['bootOptions']) {
        // data not null
        _InstanceBootOptions["default"].validateJSON(data['bootOptions']);
      }
      // validate the optional field `device`
      if (data['device']) {
        // data not null
        _Model["default"].validateJSON(data['device']);
      }
      return true;
    }
  }]);
  return InstanceCreateOptions;
}();
InstanceCreateOptions.RequiredProperties = ["flavor", "project", "os"];

/**
 * the name of the device
 * @member {String} name
 */
InstanceCreateOptions.prototype['name'] = undefined;

/**
 * Key used to encrypt the Instance
 * @member {String} key
 */
InstanceCreateOptions.prototype['key'] = undefined;

/**
 * the flavor id
 * @member {String} flavor
 */
InstanceCreateOptions.prototype['flavor'] = undefined;

/**
 * project UUID
 * @member {String} project
 */
InstanceCreateOptions.prototype['project'] = undefined;

/**
 * OS Version
 * @member {String} os
 */
InstanceCreateOptions.prototype['os'] = undefined;

/**
 * OS Build
 * @member {String} osbuild
 */
InstanceCreateOptions.prototype['osbuild'] = undefined;

/**
 * list of patches to apply
 * @member {Array.<String>} patches
 */
InstanceCreateOptions.prototype['patches'] = undefined;

/**
 * URL or image id
 * @member {String} fwpackage
 */
InstanceCreateOptions.prototype['fwpackage'] = undefined;

/**
 * URL that firmware package used to create this instance is available at
 * @member {String} origFwPackageUrl
 */
InstanceCreateOptions.prototype['origFwPackageUrl'] = undefined;

/**
 * 
 * @member {Boolean} encrypt
 */
InstanceCreateOptions.prototype['encrypt'] = undefined;

/**
 * 
 * @member {String} overrideWifiMAC
 */
InstanceCreateOptions.prototype['overrideWifiMAC'] = undefined;

/**
 * @member {module:model/VolumeOptions} volume
 */
InstanceCreateOptions.prototype['volume'] = undefined;

/**
 * Snapshot ID for this instance to be cloned from if defined
 * @member {String} snapshot
 */
InstanceCreateOptions.prototype['snapshot'] = undefined;

/**
 * @member {module:model/InstanceBootOptions} bootOptions
 */
InstanceCreateOptions.prototype['bootOptions'] = undefined;

/**
 * @member {module:model/Model} device
 */
InstanceCreateOptions.prototype['device'] = undefined;
var _default = InstanceCreateOptions;
exports["default"] = _default;
