// @corellium/client-api/dist/model/InstanceBootOptions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _InstanceBootOptionsAdditionalTag = _interopRequireDefault(require("./InstanceBootOptionsAdditionalTag"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The InstanceBootOptions model module.
 * @module model/InstanceBootOptions
 * @version 0.3.2
 */
var InstanceBootOptions = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InstanceBootOptions</code>.
   * 
   * @alias module:model/InstanceBootOptions
   */
  function InstanceBootOptions() {
    _classCallCheck(this, InstanceBootOptions);
    InstanceBootOptions.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(InstanceBootOptions, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>InstanceBootOptions</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InstanceBootOptions} obj Optional instance to populate.
     * @return {module:model/InstanceBootOptions} The populated <code>InstanceBootOptions</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InstanceBootOptions();
        if (data.hasOwnProperty('bootArgs')) {
          obj['bootArgs'] = _ApiClient["default"].convertToType(data['bootArgs'], 'String');
        }
        if (data.hasOwnProperty('restoreBootArgs')) {
          obj['restoreBootArgs'] = _ApiClient["default"].convertToType(data['restoreBootArgs'], 'String');
        }
        if (data.hasOwnProperty('udid')) {
          obj['udid'] = _ApiClient["default"].convertToType(data['udid'], 'String');
        }
        if (data.hasOwnProperty('ecid')) {
          obj['ecid'] = _ApiClient["default"].convertToType(data['ecid'], 'String');
        }
        if (data.hasOwnProperty('randomSeed')) {
          obj['randomSeed'] = _ApiClient["default"].convertToType(data['randomSeed'], 'String');
        }
        if (data.hasOwnProperty('pac')) {
          obj['pac'] = _ApiClient["default"].convertToType(data['pac'], 'Boolean');
        }
        if (data.hasOwnProperty('aprr')) {
          obj['aprr'] = _ApiClient["default"].convertToType(data['aprr'], 'Boolean');
        }
        if (data.hasOwnProperty('additionalTags')) {
          obj['additionalTags'] = _ApiClient["default"].convertToType(data['additionalTags'], [_InstanceBootOptionsAdditionalTag["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>InstanceBootOptions</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>InstanceBootOptions</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['bootArgs'] && !(typeof data['bootArgs'] === 'string' || data['bootArgs'] instanceof String)) {
        throw new Error("Expected the field `bootArgs` to be a primitive type in the JSON string but got " + data['bootArgs']);
      }
      // ensure the json data is a string
      if (data['restoreBootArgs'] && !(typeof data['restoreBootArgs'] === 'string' || data['restoreBootArgs'] instanceof String)) {
        throw new Error("Expected the field `restoreBootArgs` to be a primitive type in the JSON string but got " + data['restoreBootArgs']);
      }
      // ensure the json data is a string
      if (data['udid'] && !(typeof data['udid'] === 'string' || data['udid'] instanceof String)) {
        throw new Error("Expected the field `udid` to be a primitive type in the JSON string but got " + data['udid']);
      }
      // ensure the json data is a string
      if (data['ecid'] && !(typeof data['ecid'] === 'string' || data['ecid'] instanceof String)) {
        throw new Error("Expected the field `ecid` to be a primitive type in the JSON string but got " + data['ecid']);
      }
      // ensure the json data is a string
      if (data['randomSeed'] && !(typeof data['randomSeed'] === 'string' || data['randomSeed'] instanceof String)) {
        throw new Error("Expected the field `randomSeed` to be a primitive type in the JSON string but got " + data['randomSeed']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['additionalTags'])) {
        throw new Error("Expected the field `additionalTags` to be an array in the JSON data but got " + data['additionalTags']);
      }
      return true;
    }
  }]);
  return InstanceBootOptions;
}();
/**
 * 
 * @member {String} bootArgs
 */
InstanceBootOptions.prototype['bootArgs'] = undefined;

/**
 * 
 * @member {String} restoreBootArgs
 */
InstanceBootOptions.prototype['restoreBootArgs'] = undefined;

/**
 * Boot udid
 * @member {String} udid
 */
InstanceBootOptions.prototype['udid'] = undefined;

/**
 * Assigned ECID
 * @member {String} ecid
 */
InstanceBootOptions.prototype['ecid'] = undefined;

/**
 * Random seed to provide to boot if any
 * @member {String} randomSeed
 */
InstanceBootOptions.prototype['randomSeed'] = undefined;

/**
 * Enable PAC
 * @member {Boolean} pac
 */
InstanceBootOptions.prototype['pac'] = undefined;

/**
 * Enable APRR
 * @member {Boolean} aprr
 */
InstanceBootOptions.prototype['aprr'] = undefined;

/**
 * 
 * @member {Array.<module:model/InstanceBootOptionsAdditionalTag>} additionalTags
 */
InstanceBootOptions.prototype['additionalTags'] = undefined;
var _default = InstanceBootOptions;
exports["default"] = _default;
