// @corellium/client-api/dist/model/InstanceInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _TextInput = _interopRequireDefault(require("./TextInput"));
var _TouchCurveInput = _interopRequireDefault(require("./TouchCurveInput"));
var _TouchInput = _interopRequireDefault(require("./TouchInput"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The InstanceInput model module.
 * @module model/InstanceInput
 * @version 0.3.2
 */
var InstanceInput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InstanceInput</code>.
   * 
   * @alias module:model/InstanceInput
   * @param {(module:model/TextInput|module:model/TouchCurveInput|module:model/TouchInput)} instance The actual instance to initialize InstanceInput.
   */
  function InstanceInput() {
    var instance = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : null;
    _classCallCheck(this, InstanceInput);
    /**
     * Returns the JSON representation of the actual instance.
     * @return {string}
     */
    _defineProperty(this, "toJSON", function () {
      return this.getActualInstance();
    });
    if (instance === null) {
      this.actualInstance = null;
      return;
    }
    var match = 0;
    var errorMessages = [];
    try {
      if (typeof instance === "TouchInput") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _TouchInput["default"].validateJSON(instance); // throw an exception if no match
        // create TouchInput from JS object
        this.actualInstance = _TouchInput["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into TouchInput
      errorMessages.push("Failed to construct TouchInput: " + err);
    }
    try {
      if (typeof instance === "TouchCurveInput") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _TouchCurveInput["default"].validateJSON(instance); // throw an exception if no match
        // create TouchCurveInput from JS object
        this.actualInstance = _TouchCurveInput["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into TouchCurveInput
      errorMessages.push("Failed to construct TouchCurveInput: " + err);
    }
    try {
      if (typeof instance === "TextInput") {
        this.actualInstance = instance;
      } else {
        // plain JS object
        // validate the object
        _TextInput["default"].validateJSON(instance); // throw an exception if no match
        // create TextInput from JS object
        this.actualInstance = _TextInput["default"].constructFromObject(instance);
      }
      match++;
    } catch (err) {
      // json data failed to deserialize into TextInput
      errorMessages.push("Failed to construct TextInput: " + err);
    }
    if (match > 1) {
      throw new Error("Multiple matches found constructing `InstanceInput` with oneOf schemas TextInput, TouchCurveInput, TouchInput. Input: " + JSON.stringify(instance));
    } else if (match === 0) {
      this.actualInstance = null; // clear the actual instance in case there are multiple matches
      throw new Error("No match found constructing `InstanceInput` with oneOf schemas TextInput, TouchCurveInput, TouchInput. Details: " + errorMessages.join(", "));
    } else {// only 1 match
      // the input is valid
    }
  }

  /**
   * Constructs a <code>InstanceInput</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/InstanceInput} obj Optional instance to populate.
   * @return {module:model/InstanceInput} The populated <code>InstanceInput</code> instance.
   */
  _createClass(InstanceInput, [{
    key: "getActualInstance",
    value:
    /**
     * Gets the actual instance, which can be <code>TextInput</code>, <code>TouchCurveInput</code>, <code>TouchInput</code>.
     * @return {(module:model/TextInput|module:model/TouchCurveInput|module:model/TouchInput)} The actual instance.
     */
    function getActualInstance() {
      return this.actualInstance;
    }

    /**
     * Sets the actual instance, which can be <code>TextInput</code>, <code>TouchCurveInput</code>, <code>TouchInput</code>.
     * @param {(module:model/TextInput|module:model/TouchCurveInput|module:model/TouchInput)} obj The actual instance.
     */
  }, {
    key: "setActualInstance",
    value: function setActualInstance(obj) {
      this.actualInstance = InstanceInput.constructFromObject(obj).getActualInstance();
    }
  }], [{
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      return new InstanceInput(data);
    }
  }]);
  return InstanceInput;
}();
/**
 * Finger Positions
 * @member {Object} position
 */
/**
 * Create an instance of InstanceInput from a JSON string.
 * @param {string} json_string JSON string.
 * @return {module:model/InstanceInput} An instance of InstanceInput.
 */
_defineProperty(InstanceInput, "fromJSON", function (json_string) {
  return InstanceInput.constructFromObject(JSON.parse(json_string));
});
InstanceInput.prototype['position'] = undefined;

/**
 * Finger Positions
 * @member {Object} start
 */
InstanceInput.prototype['start'] = undefined;

/**
 * Finger Positions
 * @member {Object} end
 */
InstanceInput.prototype['end'] = undefined;

/**
 * text to type
 * @member {String} text
 */
InstanceInput.prototype['text'] = undefined;
InstanceInput.OneOf = ["TextInput", "TouchCurveInput", "TouchInput"];
var _default = InstanceInput;
exports["default"] = _default;
