// @corellium/client-api/dist/model/NetdumpFilter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The NetdumpFilter model module.
 * @module model/NetdumpFilter
 * @version 0.3.2
 */
var NetdumpFilter = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NetdumpFilter</code>.
   * 
   * @alias module:model/NetdumpFilter
   */
  function NetdumpFilter() {
    _classCallCheck(this, NetdumpFilter);
    NetdumpFilter.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(NetdumpFilter, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>NetdumpFilter</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NetdumpFilter} obj Optional instance to populate.
     * @return {module:model/NetdumpFilter} The populated <code>NetdumpFilter</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NetdumpFilter();
        if (data.hasOwnProperty('portRanges')) {
          obj['portRanges'] = _ApiClient["default"].convertToType(data['portRanges'], ['String']);
        }
        if (data.hasOwnProperty('srcPorts')) {
          obj['srcPorts'] = _ApiClient["default"].convertToType(data['srcPorts'], ['String']);
        }
        if (data.hasOwnProperty('dstPorts')) {
          obj['dstPorts'] = _ApiClient["default"].convertToType(data['dstPorts'], ['String']);
        }
        if (data.hasOwnProperty('ports')) {
          obj['ports'] = _ApiClient["default"].convertToType(data['ports'], ['String']);
        }
        if (data.hasOwnProperty('protocols')) {
          obj['protocols'] = _ApiClient["default"].convertToType(data['protocols'], ['String']);
        }
        if (data.hasOwnProperty('processes')) {
          obj['processes'] = _ApiClient["default"].convertToType(data['processes'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>NetdumpFilter</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>NetdumpFilter</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['portRanges'])) {
        throw new Error("Expected the field `portRanges` to be an array in the JSON data but got " + data['portRanges']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['srcPorts'])) {
        throw new Error("Expected the field `srcPorts` to be an array in the JSON data but got " + data['srcPorts']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['dstPorts'])) {
        throw new Error("Expected the field `dstPorts` to be an array in the JSON data but got " + data['dstPorts']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['ports'])) {
        throw new Error("Expected the field `ports` to be an array in the JSON data but got " + data['ports']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['protocols'])) {
        throw new Error("Expected the field `protocols` to be an array in the JSON data but got " + data['protocols']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['processes'])) {
        throw new Error("Expected the field `processes` to be an array in the JSON data but got " + data['processes']);
      }
      return true;
    }
  }]);
  return NetdumpFilter;
}();
/**
 * 
 * @member {Array.<String>} portRanges
 */
NetdumpFilter.prototype['portRanges'] = undefined;

/**
 * 
 * @member {Array.<String>} srcPorts
 */
NetdumpFilter.prototype['srcPorts'] = undefined;

/**
 * 
 * @member {Array.<String>} dstPorts
 */
NetdumpFilter.prototype['dstPorts'] = undefined;

/**
 * 
 * @member {Array.<String>} ports
 */
NetdumpFilter.prototype['ports'] = undefined;

/**
 * 
 * @member {Array.<String>} protocols
 */
NetdumpFilter.prototype['protocols'] = undefined;

/**
 * 
 * @member {Array.<String>} processes
 */
NetdumpFilter.prototype['processes'] = undefined;
var _default = NetdumpFilter;
exports["default"] = _default;
