// @corellium/client-api/dist/model/PasswordChangeBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The PasswordChangeBody model module.
 * @module model/PasswordChangeBody
 * @version 0.3.2
 */
var PasswordChangeBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PasswordChangeBody</code>.
   * 
   * @alias module:model/PasswordChangeBody
   * @param user {String} userId
   * @param oldPassword {String} old password
   * @param newPassword {String} new password
   */
  function PasswordChangeBody(user, oldPassword, newPassword) {
    _classCallCheck(this, PasswordChangeBody);
    PasswordChangeBody.initialize(this, user, oldPassword, newPassword);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PasswordChangeBody, null, [{
    key: "initialize",
    value: function initialize(obj, user, oldPassword, newPassword) {
      obj['user'] = user;
      obj['old-password'] = oldPassword;
      obj['new-password'] = newPassword;
    }

    /**
     * Constructs a <code>PasswordChangeBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PasswordChangeBody} obj Optional instance to populate.
     * @return {module:model/PasswordChangeBody} The populated <code>PasswordChangeBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PasswordChangeBody();
        if (data.hasOwnProperty('user')) {
          obj['user'] = _ApiClient["default"].convertToType(data['user'], 'String');
        }
        if (data.hasOwnProperty('old-password')) {
          obj['old-password'] = _ApiClient["default"].convertToType(data['old-password'], 'String');
        }
        if (data.hasOwnProperty('new-password')) {
          obj['new-password'] = _ApiClient["default"].convertToType(data['new-password'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PasswordChangeBody</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PasswordChangeBody</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(PasswordChangeBody.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['user'] && !(typeof data['user'] === 'string' || data['user'] instanceof String)) {
        throw new Error("Expected the field `user` to be a primitive type in the JSON string but got " + data['user']);
      }
      // ensure the json data is a string
      if (data['old-password'] && !(typeof data['old-password'] === 'string' || data['old-password'] instanceof String)) {
        throw new Error("Expected the field `old-password` to be a primitive type in the JSON string but got " + data['old-password']);
      }
      // ensure the json data is a string
      if (data['new-password'] && !(typeof data['new-password'] === 'string' || data['new-password'] instanceof String)) {
        throw new Error("Expected the field `new-password` to be a primitive type in the JSON string but got " + data['new-password']);
      }
      return true;
    }
  }]);
  return PasswordChangeBody;
}();
PasswordChangeBody.RequiredProperties = ["user", "old-password", "new-password"];

/**
 * userId
 * @member {String} user
 */
PasswordChangeBody.prototype['user'] = undefined;

/**
 * old password
 * @member {String} old-password
 */
PasswordChangeBody.prototype['old-password'] = undefined;

/**
 * new password
 * @member {String} new-password
 */
PasswordChangeBody.prototype['new-password'] = undefined;
var _default = PasswordChangeBody;
exports["default"] = _default;
