// @corellium/client-api/dist/model/ProxyConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ProxyConfig model module.
 * @module model/ProxyConfig
 * @version 0.3.2
 */
var ProxyConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProxyConfig</code>.
   * Represents a Proxy configuration object.
   * @alias module:model/ProxyConfig
   */
  function ProxyConfig() {
    _classCallCheck(this, ProxyConfig);
    ProxyConfig.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ProxyConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ProxyConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProxyConfig} obj Optional instance to populate.
     * @return {module:model/ProxyConfig} The populated <code>ProxyConfig</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProxyConfig();
        if (data.hasOwnProperty('devicePort')) {
          obj['devicePort'] = _ApiClient["default"].convertToType(data['devicePort'], 'Number');
        }
        if (data.hasOwnProperty('firstAvailable')) {
          obj['firstAvailable'] = _ApiClient["default"].convertToType(data['firstAvailable'], 'Boolean');
        }
        if (data.hasOwnProperty('expose')) {
          obj['expose'] = _ApiClient["default"].convertToType(data['expose'], 'Boolean');
        }
        if (data.hasOwnProperty('routerPort')) {
          obj['routerPort'] = _ApiClient["default"].convertToType(data['routerPort'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ProxyConfig</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ProxyConfig</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);
  return ProxyConfig;
}();
/**
 * The device port to use for proxying.
 * @member {Number} devicePort
 */
ProxyConfig.prototype['devicePort'] = undefined;

/**
 * If `true`, the first available port will be used if `devicePort` is not available.
 * @member {Boolean} firstAvailable
 */
ProxyConfig.prototype['firstAvailable'] = undefined;

/**
 * If `true`, the proxy will be exposed to the external interface.
 * @member {Boolean} expose
 */
ProxyConfig.prototype['expose'] = undefined;

/**
 * The router port to use for proxying.
 * @member {Number} routerPort
 */
ProxyConfig.prototype['routerPort'] = undefined;
var _default = ProxyConfig;
exports["default"] = _default;
