// @corellium/client-api/dist/model/TrialRequestMetadata.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The TrialRequestMetadata model module.
 * @module model/TrialRequestMetadata
 * @version 0.3.2
 */
var TrialRequestMetadata = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TrialRequestMetadata</code>.
   * 
   * @alias module:model/TrialRequestMetadata
   */
  function TrialRequestMetadata() {
    _classCallCheck(this, TrialRequestMetadata);
    TrialRequestMetadata.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(TrialRequestMetadata, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>TrialRequestMetadata</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TrialRequestMetadata} obj Optional instance to populate.
     * @return {module:model/TrialRequestMetadata} The populated <code>TrialRequestMetadata</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TrialRequestMetadata();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('company')) {
          obj['company'] = _ApiClient["default"].convertToType(data['company'], 'String');
        }
        if (data.hasOwnProperty('malware')) {
          obj['malware'] = _ApiClient["default"].convertToType(data['malware'], 'Boolean');
        }
        if (data.hasOwnProperty('research')) {
          obj['research'] = _ApiClient["default"].convertToType(data['research'], 'Boolean');
        }
        if (data.hasOwnProperty('pentest')) {
          obj['pentest'] = _ApiClient["default"].convertToType(data['pentest'], 'Boolean');
        }
        if (data.hasOwnProperty('other')) {
          obj['other'] = _ApiClient["default"].convertToType(data['other'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>TrialRequestMetadata</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TrialRequestMetadata</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['company'] && !(typeof data['company'] === 'string' || data['company'] instanceof String)) {
        throw new Error("Expected the field `company` to be a primitive type in the JSON string but got " + data['company']);
      }
      // ensure the json data is a string
      if (data['other'] && !(typeof data['other'] === 'string' || data['other'] instanceof String)) {
        throw new Error("Expected the field `other` to be a primitive type in the JSON string but got " + data['other']);
      }
      return true;
    }
  }]);
  return TrialRequestMetadata;
}();
/**
 * 
 * @member {String} name
 */
TrialRequestMetadata.prototype['name'] = undefined;

/**
 * provided company name
 * @member {String} company
 */
TrialRequestMetadata.prototype['company'] = undefined;

/**
 * option from sign up form, interesting using for malware
 * @member {Boolean} malware
 */
TrialRequestMetadata.prototype['malware'] = undefined;

/**
 * option from sign up form, interesting using for research
 * @member {Boolean} research
 */
TrialRequestMetadata.prototype['research'] = undefined;

/**
 * option from sign up form, interesting using for pentesting
 * @member {Boolean} pentest
 */
TrialRequestMetadata.prototype['pentest'] = undefined;

/**
 * option from sign up form, interesting using for other area added here
 * @member {String} other
 */
TrialRequestMetadata.prototype['other'] = undefined;
var _default = TrialRequestMetadata;
exports["default"] = _default;
