// @corellium/client-api/dist/model/V1CreateHookParameters.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The V1CreateHookParameters model module.
 * @module model/V1CreateHookParameters
 * @version 0.3.2
 */
var V1CreateHookParameters = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>V1CreateHookParameters</code>.
   * 
   * @alias module:model/V1CreateHookParameters
   * @param label {String} Label
   * @param address {String} Address
   * @param patch {String} Patch
   * @param patchType {module:model/V1CreateHookParameters.PatchTypeEnum} Patch Type
   */
  function V1CreateHookParameters(label, address, patch, patchType) {
    _classCallCheck(this, V1CreateHookParameters);
    V1CreateHookParameters.initialize(this, label, address, patch, patchType);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(V1CreateHookParameters, null, [{
    key: "initialize",
    value: function initialize(obj, label, address, patch, patchType) {
      obj['label'] = label;
      obj['address'] = address;
      obj['patch'] = patch;
      obj['patchType'] = patchType;
    }

    /**
     * Constructs a <code>V1CreateHookParameters</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/V1CreateHookParameters} obj Optional instance to populate.
     * @return {module:model/V1CreateHookParameters} The populated <code>V1CreateHookParameters</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new V1CreateHookParameters();
        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'String');
        }
        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
        }
        if (data.hasOwnProperty('patch')) {
          obj['patch'] = _ApiClient["default"].convertToType(data['patch'], 'String');
        }
        if (data.hasOwnProperty('patchType')) {
          obj['patchType'] = _ApiClient["default"].convertToType(data['patchType'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>V1CreateHookParameters</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>V1CreateHookParameters</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(V1CreateHookParameters.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['label'] && !(typeof data['label'] === 'string' || data['label'] instanceof String)) {
        throw new Error("Expected the field `label` to be a primitive type in the JSON string but got " + data['label']);
      }
      // ensure the json data is a string
      if (data['address'] && !(typeof data['address'] === 'string' || data['address'] instanceof String)) {
        throw new Error("Expected the field `address` to be a primitive type in the JSON string but got " + data['address']);
      }
      // ensure the json data is a string
      if (data['patch'] && !(typeof data['patch'] === 'string' || data['patch'] instanceof String)) {
        throw new Error("Expected the field `patch` to be a primitive type in the JSON string but got " + data['patch']);
      }
      // ensure the json data is a string
      if (data['patchType'] && !(typeof data['patchType'] === 'string' || data['patchType'] instanceof String)) {
        throw new Error("Expected the field `patchType` to be a primitive type in the JSON string but got " + data['patchType']);
      }
      return true;
    }
  }]);
  return V1CreateHookParameters;
}();
V1CreateHookParameters.RequiredProperties = ["label", "address", "patch", "patchType"];

/**
 * Label
 * @member {String} label
 */
V1CreateHookParameters.prototype['label'] = undefined;

/**
 * Address
 * @member {String} address
 */
V1CreateHookParameters.prototype['address'] = undefined;

/**
 * Patch
 * @member {String} patch
 */
V1CreateHookParameters.prototype['patch'] = undefined;

/**
 * Patch Type
 * @member {module:model/V1CreateHookParameters.PatchTypeEnum} patchType
 */
V1CreateHookParameters.prototype['patchType'] = undefined;

/**
 * Allowed values for the <code>patchType</code> property.
 * @enum {String}
 * @readonly
 */
V1CreateHookParameters['PatchTypeEnum'] = {
  /**
   * value: "csmfcc"
   * @const
   */
  "csmfcc": "csmfcc",
  /**
   * value: "csmfvm"
   * @const
   */
  "csmfvm": "csmfvm"
};
var _default = V1CreateHookParameters;
exports["default"] = _default;
