// @corellium/client-api/dist/model/WebPlayerCreateSessionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Features = _interopRequireDefault(require("./Features"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Corellium API
                                                                                                                                                                                                                                                                                                                                                                                               * REST API to manage your virtual devices.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.5.0-16775
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The WebPlayerCreateSessionRequest model module.
 * @module model/WebPlayerCreateSessionRequest
 * @version 0.3.2
 */
var WebPlayerCreateSessionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebPlayerCreateSessionRequest</code>.
   * 
   * @alias module:model/WebPlayerCreateSessionRequest
   * @param projectId {String} Project Identifier
   * @param instanceId {String} VM Instance Identifier
   * @param expiresIn {Number} Number of seconds token remains valid
   * @param features {module:model/Features} 
   */
  function WebPlayerCreateSessionRequest(projectId, instanceId, expiresIn, features) {
    _classCallCheck(this, WebPlayerCreateSessionRequest);
    WebPlayerCreateSessionRequest.initialize(this, projectId, instanceId, expiresIn, features);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(WebPlayerCreateSessionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, projectId, instanceId, expiresIn, features) {
      obj['projectId'] = projectId;
      obj['instanceId'] = instanceId;
      obj['expiresIn'] = expiresIn;
      obj['features'] = features;
    }

    /**
     * Constructs a <code>WebPlayerCreateSessionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebPlayerCreateSessionRequest} obj Optional instance to populate.
     * @return {module:model/WebPlayerCreateSessionRequest} The populated <code>WebPlayerCreateSessionRequest</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebPlayerCreateSessionRequest();
        if (data.hasOwnProperty('projectId')) {
          obj['projectId'] = _ApiClient["default"].convertToType(data['projectId'], 'String');
        }
        if (data.hasOwnProperty('instanceId')) {
          obj['instanceId'] = _ApiClient["default"].convertToType(data['instanceId'], 'String');
        }
        if (data.hasOwnProperty('expiresIn')) {
          obj['expiresIn'] = _ApiClient["default"].convertToType(data['expiresIn'], 'Number');
        }
        if (data.hasOwnProperty('features')) {
          obj['features'] = _Features["default"].constructFromObject(data['features']);
        }
        if (data.hasOwnProperty('clientId')) {
          obj['clientId'] = _ApiClient["default"].convertToType(data['clientId'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>WebPlayerCreateSessionRequest</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>WebPlayerCreateSessionRequest</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(WebPlayerCreateSessionRequest.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['projectId'] && !(typeof data['projectId'] === 'string' || data['projectId'] instanceof String)) {
        throw new Error("Expected the field `projectId` to be a primitive type in the JSON string but got " + data['projectId']);
      }
      // ensure the json data is a string
      if (data['instanceId'] && !(typeof data['instanceId'] === 'string' || data['instanceId'] instanceof String)) {
        throw new Error("Expected the field `instanceId` to be a primitive type in the JSON string but got " + data['instanceId']);
      }
      // validate the optional field `features`
      if (data['features']) {
        // data not null
        _Features["default"].validateJSON(data['features']);
      }
      // ensure the json data is a string
      if (data['clientId'] && !(typeof data['clientId'] === 'string' || data['clientId'] instanceof String)) {
        throw new Error("Expected the field `clientId` to be a primitive type in the JSON string but got " + data['clientId']);
      }
      return true;
    }
  }]);
  return WebPlayerCreateSessionRequest;
}();
WebPlayerCreateSessionRequest.RequiredProperties = ["projectId", "instanceId", "expiresIn", "features"];

/**
 * Project Identifier
 * @member {String} projectId
 */
WebPlayerCreateSessionRequest.prototype['projectId'] = undefined;

/**
 * VM Instance Identifier
 * @member {String} instanceId
 */
WebPlayerCreateSessionRequest.prototype['instanceId'] = undefined;

/**
 * Number of seconds token remains valid
 * @member {Number} expiresIn
 */
WebPlayerCreateSessionRequest.prototype['expiresIn'] = undefined;

/**
 * @member {module:model/Features} features
 */
WebPlayerCreateSessionRequest.prototype['features'] = undefined;

/**
 * Optional string value supplied by client for tracking purposes
 * @member {String} clientId
 */
WebPlayerCreateSessionRequest.prototype['clientId'] = undefined;
var _default = WebPlayerCreateSessionRequest;
exports["default"] = _default;
