// @corellium/corellium-cli/src/commands/images/create.js
const { handleError } = require('../../error');
const fs = require('fs');
const { displayTable } = require('../../table');
const { getApi } = require('../../utils');

async function builder (yargs) {
  yargs
    .option('project', {
      type: 'input',
      describe: 'Project UUID',
      string: true,
    })
    .option('instance', {
      type: 'input',
      describe: 'Instance UUID',
      string: true,
    })
    .option('format', {
      type: 'input',
      describe: 'Output format (default is json)',
      string: true,
      choices: ['table', 'json', 'csv'],
      default: 'json',
    })
    .option('verbose', {
      alias: 'v',
      type: 'boolean',
      describe: 'Console will receive verbose error output',
    })
    .positional('imageName', {
      type: 'string',
      describe: 'Image name',
      demandOption: true,
    })
    .positional('imageType', {
      type: 'string',
      describe: 'Image type to upload',
      demandOption: true,
    })
    .positional('encoding', {
      type: 'string',
      describe: 'Encoding',
      choices: ['plain', 'encrypted'],
    })
    .positional('file', {
      type: 'string',
      describe: 'Path to image file',
      demandOption: true,
    })
    .check((argv) => {
      if (!argv.project && !argv.instance) {
        throw new Error('You must specify either the project or instance UUID');
      }
      return true;
    });
}

async function handler (argv) {
  const imageType = argv.imageType;
  const imagePath = argv.file;
  const encoding = argv.encoding;
  const imageName = argv.imageName;
  const project = argv.project;
  const instance = argv.instance;
  const api = await getApi();

  const opts = {
    name: imageName,
    project: project,
    instance: instance,
    file: fs.createReadStream(imagePath),
  };

  try {
    const image = await api.v1CreateImage(imageType, encoding, opts);
    displayTable(argv.format, [image], ['status', 'id', 'name', 'type', 'project']);
  } catch (err) {
    handleError(err, 'Failed to create image', argv.verbose);
  }
}

module.exports = {
  builder,
  handler,
  command: 'create <imageName> <imageType> <encoding> <file>',
  describe:
    'Create image and associate it with a project or instance (you must specify either the project or instance UUID)',
};
